src/haxelib/client/RepoManager.hx

```diff
@@ -0,0 +1,165 @@
+package haxelib.client;
+
+import sys.FileSystem;
+import sys.io.File;
+import haxe.io.Path;
+
+import haxelib.client.FsUtils.*;
+
+using StringTools;
+
+class RepoException extends haxe.Exception {}
```

Comment:
`haxe.Exception` is not available prior to Haxe 4.1.
I think a check like this would be good for users with older Haxe:
```haxe
#if (haxe_ver < 4.1)
  #error "RepoManager requires Haxe 4.1 or newer"
#end
```

Reply:
Instead of that I just changed them to raw exceptions, because otherwise haxelib wouldn't be usable at all with the older versions.

Reply:
I'm still not sure why exactly are we trying to keep haxelib compileable with older haxe versions. We never really update it anyway, and since Haxe just runs haxelib executable, it also doesn't care which version was it compiled with...

Reply:
That's true, when running it has a haxelib `haxelib run haxelib` as long as it's running using the run.n file there shouldn't be a problem (as opposed to Run.hx, which it doesn't right now).

Perhaps a separate issue should be opened regarding haxe 3 compatibility though?

Reply:
If we want to promote it as a library, then the source code will be used instead of `run.n`.
I'm ok with breaking backwards compatibility, but we can emit a user-friendly error in this case for free. So why not?

Reply:
Added the warning